flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/RecordWriterOutput.java

```diff
@@ -110,12 +108,14 @@ public void collect(StreamRecord<OUT> record) {
 
     @Override
     public void emitWatermark(Watermark mark) {
-        // watermark could've been generated somewhere in the pipeline even though an IDLE status
-        // was emitted. It might've originated from a periodic watermark generator or just a wrong
-        // behaving operator
-        try (AutoCloseable ignored = announcedStatus.ensureActive(this::writeStreamStatus)) {
-            watermarkGauge.setCurrentWatermark(mark.getTimestamp());
-            serializationDelegate.setInstance(mark);
+        if (announcedStatus.isIdle()) {
+            return;
+        }
```

Comment:
This is as it it in 1.13 - emitting watermarks do not switch the status to active. But that's something that we might want to change in the future?

Reply:
Yes, in 1.13 we ignore/swallow watermarks emitted in the `IDLE`state. There we check the status via `StreamStatusProvider` though.

Reply:
The comment about tests reminded me that I dropped a test for it along with the behaviour in outputs.